DependencyInjection/SmartboxIntegrationFrameworkExtension.php

```diff
@@ -226,49 +225,44 @@ protected function loadQueueDrivers(ContainerBuilder $container)
+                        $container->findDefinition('smartesb.protocols.queue')
+                            ->addMethodCall('setDefaultConsumer', [new Reference('smartesb.consumers.queue')]);
+                    }
+
                     break;
 
                 case 'amqp':
-                    if (!class_exists('AMQPConnection')) {
-                        throw new \RuntimeException('You need the amqp extension to use AMQP driver.');
-                    }
-
-                    if (empty($driverConfig['connections'] ?? [])) {
-                        throw new \InvalidArgumentException('You need to define at least one connection to use the AMQP driver.');
+                    if (!class_exists(\PhpAmqpLib\Package::class)) {
+                        throw new \LogicException('To use the AMQP driver, you\'ll need the php-amqplib library. Please run \'composer require php-amqplib/php-amqplib\' to install it.');
                     }
 
-                    $queueManager = $container->findDefinition('smartesb.amqp.queue_manager');
-                    foreach ($driverConfig['connections'] as $index => $uri) {
-                        $connection = parse_url($uri);
-                        $connectionId = "$driverId.connection.$index";
-
-                        $container->register($connectionId, 'AMQPConnection')->setArguments([[
-                            'host' => $connection['host'],
-                            'port' => $connection['port'] ?? 5672,
-                            'vhost' => trim($connection['path'] ?? '', '/'),
-                            'login' => $connection['user'],
-                            'password' => $connection['pass'],
-                        ]]);
+                    $driverDef = new Definition(PhpAmqpLibDriver::class);
+                    $driverDef->addMethodCall('configure', [
+                        $driverConfig['host'],
+                        $driverConfig['username'],
+                        $driverConfig['password'],
+                        $driverConfig['vhost']
+                    ]);
 
-                        $queueManager->addMethodCall('addConnection', [new Reference($connectionId)]);
-                    }
-                    $this->useAmqp = true;
+                    $queueDriverRegistry->addMethodCall('setDriver', [$driverName, $driverDef]);
 
-                    $driverDef = $container->findDefinition('smartesb.drivers.queue.amqp');
                     $driverDef->addMethodCall('setId', [$driverId]);
-                    $driverDef->addMethodCall('configure', [null, null, null, $driverConfig['format']]);
-
-                    $queueDriverRegistry->addMethodCall('setDriver', [$driverName, new Reference($driverId)]);
+                    $driverDef->addMethodCall('setSerializer', [new Reference('jms_serializer')]);
+                    $driverDef->addMethodCall('setFormat', [$driverConfig['format']]);
```

Comment:
Add `$driverDef->addMethodCall('setMessageFactory', [new Reference('smartesb.message_factory')]);` once `smartesb.drivers.queue.amqp` is removed from integration-framework-bundle/Resources/config/services.yml

Reply:
Done.